Source/Core/DolphinQt/Config/ARCodeWidget.cpp

```diff
@@ -114,6 +115,13 @@ void ARCodeWidget::OnItemChanged(QListWidgetItem* item)
 {
   m_ar_codes[m_code_list->row(item)].enabled = (item->checkState() == Qt::Checked);
 
+#ifdef USE_RETRO_ACHIEVEMENTS
+  {
+    std::lock_guard lg{AchievementManager::GetInstance().GetLock()};
+    AchievementManager::GetInstance().FilterApprovedARCodes(m_ar_codes, m_game_id);
```

Comment:
Doing filtering in ARCodeWidget and GeckoCodeWidget looks a lot to me like it would cause the problem AdmiralCurtiss mentioned. Look just a few lines below and you'll see a call to SaveCodes.